ql/src/java/org/apache/hadoop/hive/ql/session/ClearDanglingScratchDir.java

```diff
@@ -236,4 +243,29 @@ static Options createOptions() {
 
     return result;
   }
+
+  /**
+   * While deleting dangling scratch dirs from hdfs, we can clean corresponding local files as well
+   * @param sessionName prefix to determine removable tmp files
+   * @param localTmpdir local tmp file location
+   */
+  public void removeLocalTmpFiles(String sessionName, String localTmpdir) {
```

Comment:
@veghlaci05 this is visible for testing. I have added `@VisibleForTesting` annotation. Initially the method was private and in unit test I have executed whole service, but service failed with below exception;
```
java.lang.UnsupportedOperationException: Append is not supported by ChecksumFileSystem
```
at below [point](https://github.com/apache/hive/blob/badba5e8a3b98f8c7be4903a6d9b62f8ea5e142e/ql/src/java/org/apache/hadoop/hive/ql/session/ClearDanglingScratchDir.java#L142);
```
IOUtils.closeStream(fs.append(lockFilePath));
```

Reply:
You should discover and enhance org.apache.hadoop.hive.ql.session.TestClearDanglingScratchDir instead of writing a test in TestCleanupService which is a completely different test.

This should eliminate the need of testing the method separately, so thus making it public.

Reply:
Found the issue; `UnsupportedOperationException` has been considered in the initial solution, but the problem is that it is under `catch` block where `RemoteException` is caught, please see [here](https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/session/ClearDanglingScratchDir.java#L136-L145).

As `UnsupportedOperationException` extends `RuntimeException`, this condition was never be true.

So, I have removed the related code to separate `catch` block, and executed unit test again with whole service in which `removeLocalTmpFiles` is private, it worked well;
![image](https://github.com/apache/hive/assets/73108963/bb8be817-6f17-43f8-ac70-3fba5f21e551)